src/components/capture/useDiscoverCapture.ts

```diff
@@ -130,94 +111,15 @@ function useDiscoverCapture(
-        useResourceConfig_evaluateDiscoveredCollections();
     const resourceConfigHasErrors =
         useResourceConfig_resourceConfigErrorsExist();
-    const resetCollections = useResourceConfig_resetConfigAndCollections();
-
-    const storeDiscoveredCollections = useCallback(
-        async (newDraftId: string) => {
-            // TODO (optimization | typing): Narrow the columns selected from the draft_specs_ext table.
-            //   More columns are selected than required to appease the typing of the editor store.
-            const draftSpecsResponse = await getDraftSpecsBySpecType(
-                newDraftId,
-                entityType
-            );
-
-            if (draftSpecsResponse.error) {
-                return callFailed({
-                    error: {
-                        title: 'captureCreate.generate.failedErrorTitle',
-                        error: draftSpecsResponse.error,
-                    },
-                });
-            }
-
-            if (draftSpecsResponse.data && draftSpecsResponse.data.length > 0) {
-                resetCollections();
-
-                setDiscoveredCollections(draftSpecsResponse.data[0]);
-
-                const supabaseConfig: SupabaseConfig | null =
-                    editWorkflow && lastPubId
-                        ? { catalogName: entityName, lastPubId }
-                        : null;
 
-                const updatedDraftSpecsResponse =
-                    await modifyDiscoveredDraftSpec(
-                        draftSpecsResponse,
-                        supabaseConfig
-                    );
-
-                if (updatedDraftSpecsResponse.error) {
-                    return callFailed({
-                        error: {
-                            title: 'captureCreate.generate.failedErrorTitle',
-                            error: updatedDraftSpecsResponse.error,
-                        },
-                    });
-                }
-
-                if (
-                    updatedDraftSpecsResponse.data &&
-                    updatedDraftSpecsResponse.data.length > 0
-                ) {
-                    evaluateDiscoveredCollections(updatedDraftSpecsResponse);
-
-                    setEncryptedEndpointConfig({
-                        data: updatedDraftSpecsResponse.data[0].spec.endpoint
-                            .connector.config,
-                    });
-                }
-
-                setDraftId(newDraftId);
-                setPersistedDraftId(newDraftId);
-            }
-        },
-        [
-            callFailed,
-            editWorkflow,
-            entityName,
-            entityType,
-            evaluateDiscoveredCollections,
-            lastPubId,
-            resetCollections,
-            setDiscoveredCollections,
-            setDraftId,
-            setEncryptedEndpointConfig,
-            setPersistedDraftId,
-        ]
-    );
+    const storeDiscoveredCollections = useStoreDiscoveredCaptures();
```

Comment:
All the stuff was moved into a function. This was so the evolved schemas could get updated just like the discovered ones. Most everything was moved as it.